torchgeo/datasets/geo.py

```diff
@@ -298,8 +304,10 @@ def files(self) -> set[str]:
             if os.path.isdir(path):
                 pathname = os.path.join(path, "**", self.filename_glob)
                 files |= set(glob.iglob(pathname, recursive=True))
-            else:
+            elif os.path.isfile(path) or path_is_vsi(path):
                 files.add(path)
+            else:
+                continue
```

Comment:
This doesn't do anything and can be removed

Reply:
Do you think it would make sense to rase a warning here? e.g.

```python
warnings.warn(
    "Could not find any relevant files for provided path '{path}'.",
    UserWarning,
)
```

Reply:
Might be a good idea. We raise an error if no files were found, but we don't warn the user if multiple paths were given and only some were valid.